diff -ruN lilo-22.3.org/Makefile lilo-22.3/Makefile
--- lilo-22.3.org/Makefile	Mon May 20 13:48:52 2002
+++ lilo-22.3/Makefile	Mon May 20 13:56:22 2002
@@ -28,6 +28,8 @@
 # * LVM		  Enables support for booting from LVM partitions.  This will
 #		  OOPS if you try to use this and are using a kernel < 2.4.7
 #		  that has not been patched to at least LVM 0.9.1beta6. (new)
+# * EVMS	  Enables support for booting from EVMS volumes. Requires a
+#		  2.4 or 2.5 kernel patched to EVMS version 0.9.2 or later.
 # * M386	  Use 386 instructions in assembly code to reduce size of
 #		  second-stage loader (recommended)
 #   NO1STDIAG	  Don't show diagnostic on read errors in the first stage
@@ -54,7 +56,7 @@
 # * VERSION	  Prints version string at LILO boot prompt.
 #   XL_SECS=n	  Support for extra large (non-standard) floppies.
 
-CONFIG=-DBDATA -DBUILTIN -DDSECS=3 -DIGNORECASE -DLBA32 -DLVM -DM386 \
+CONFIG=-DBDATA -DBUILTIN -DDSECS=3 -DIGNORECASE -DLBA32 -DLVM -DEVMS -DM386 \
   -DONE_SHOT -DPASS160 -DREISERFS -DREWRITE_TABLE -DSOLO_CHAIN -DVARSETUP \
   -DVERSION -DUNIFY
 
