diff -urNp -x '*.orig' lilo-24.2.org/diagnose/Makefile lilo-24.2/diagnose/Makefile
--- lilo-24.2.org/diagnose/Makefile	2010-06-18 19:06:27.000000000 +0200
+++ lilo-24.2/diagnose/Makefile	2026-03-03 00:21:59.225285457 +0100
@@ -5,11 +5,11 @@
 include ../make.vars
 
 MODEL=s
-CC = bcc -M$(MODEL)
+BCC = bcc -M$(MODEL)
 CPP = gcc -E
 CINCL = /usr/lib/bcc/include
 AS = as86
-LINK = $(CC) -v
+LINK = $(BCC) -v
 COPT = -ansi -I$(CINCL) -c -O -DDEBUG=0
 LOPT = -m >$*.map -M -d
 AOPT = -0 -a -w
@@ -21,7 +21,7 @@ LOOP=/dev/loop5
 .SUFFIXES:  .com .S
 
 .c.o:
-	$(CC) $(COPT) -o $@ $< -A-l -A$*.lis
+	$(BCC) $(COPT) -o $@ $< -A-l -A$*.lis
 
 .o.com:
 	$(LINK) $(LOPT) -o $@ $^
