diff -ur lilo-21.5/lilo.c lilo-21.5-fixed/lilo.c
--- lilo-21.5/lilo.c	Mon Jul 17 03:03:45 2000
+++ lilo-21.5-fixed/lilo.c	Mon Jul 24 13:27:32 2000
@@ -489,6 +489,10 @@
 			  "for root if using PASSWORD\n",config_file);
 	}
     }
+    if (cfg_get_strg(cf_options,"boot") == 0) {
+	fprintf(stderr, "cannot find boot = line in config file. aborting.\n");
+	exit(1);
+    }
     preload_dev_cache();
     if (identify) identify_image(identify,ident_opt);
     if (strncmp("/dev/md",cfg_get_strg(cf_options,"boot"),7) == 0) {
