--- lilo-22.6.1/mkrescue	2004-11-16 08:50:08.000000000 +0000
+++ lilo-22.6.1.new/mkrescue	2005-01-04 22:41:53.456867448 +0000
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 #
 # mkrescue
 #
@@ -612,7 +612,7 @@
 
 
 
-pushd $mount >/dev/null 2>/dev/null
+SAVEPWD="$(pwd)"
 if [ "$fast" = fast ]; then
 	bootcmd="-b /dev/$loopback"
 fi
@@ -629,7 +629,7 @@
 else
 	$lilo -C $mountconfig $compact $bootcmd || fast=error
 fi
-popd >/dev/null 2>/dev/null
+cd "$SAVEPWD"
 if [ "$fast" = error ]; then
 	echo -n `$lilo -V`
 	echo " failure."
@@ -683,4 +683,4 @@
 
 echo "All done."
 echo ""
-exit 0
\ No newline at end of file
+exit 0
