diff -urNp -x '*.orig' lilo-24.2.org/mkrescue lilo-24.2/mkrescue
--- lilo-24.2.org/mkrescue	2011-06-21 01:22:16.000000000 +0200
+++ lilo-24.2/mkrescue	2026-03-03 00:21:42.477374881 +0100
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 #
 #       mkrescue - create a boot floppy or cd image with the current kernel
 #
@@ -624,7 +624,7 @@ echo ""
 
 
 
-pushd $mount >/dev/null 2>/dev/null
+SAVEPWD="$(pwd)"
 if [ "$fast" = fast ]; then
 	bootcmd="-b /dev/$loopback"
 fi
@@ -644,7 +644,7 @@ fi
 else
 	$lilo -C $mountconfig $compact $bootcmd || fast=error
 fi
-popd >/dev/null 2>/dev/null
+cd "$SAVEPWD"
 if [ "$fast" = error ]; then
 	echo -n `$lilo -V`
 	echo " failure."
