diff -ruN lilo-22.3/Makefile lilo-22.3.my/Makefile
--- lilo-22.3/Makefile	Fri May  3 20:16:04 2002
+++ lilo-22.3.my/Makefile	Mon May 20 13:45:53 2002
@@ -69,7 +69,7 @@
 CFG_DIR=/etc
 BOOT_DIR=/boot
 USRSBIN_DIR=/usr/sbin
-MAN_DIR=`manpath | sed "s/:.*//"`
+MAN_DIR=/usr/share/man
 BUILTIN=`./lilo -X | grep CFLAGS | grep LCF_BUILTIN | wc -l | sed "s/ //g"`
 
 PCONFIG=`( if [ -r $$ROOT/etc/lilo.defines ]; then \
