diff -urN aa/lilo-22.5.8/diagnose/Makefile lilo-22.5.8/diagnose/Makefile
--- aa/lilo-22.5.8/diagnose/Makefile	2003-03-10 23:35:14.000000000 +0100
+++ lilo-22.5.8/diagnose/Makefile	2004-02-03 13:06:47.790986632 +0100
@@ -1,11 +1,11 @@
 # Makefile for BCC tests
 #
 MODEL=s
-CC = bcc -M$(MODEL)
+BCC = bcc -M$(MODEL)
 CPP = gcc -E
 CINCL = /usr/lib/bcc/include/
 AS = as86
-LINK = $(CC) -v
+LINK = $(BCC) -v
 COPT = -ansi -I$(CINCL) -c -O -DDEBUG=0
 LOPT = -m >$*.map -M -d
 AOPT = -0 -a -w
@@ -17,7 +17,7 @@
 .SUFFIXES:	.com .S
 
 .c.o:
-	$(CC) $(COPT) -o $@ $<
+	$(BCC) $(COPT) -o $@ $<
 
 .o.com:
 	$(LINK) $(LOPT) -o $@ $^
