--- lilo-22.1/bsect.c~	Wed Oct 31 21:56:33 2001
+++ lilo-22.1/bsect.c	Tue Nov 27 21:34:43 2001
@@ -237,13 +237,12 @@
 				shsInfo.digest[j];
 #endif
 	    if(verbose >= 2) {
-		if (j==0) printf("Password "
+		if (j==0)
 #ifdef CRC_PASSWORDS
-					"CRC-32"
+		    printf("Password CRC-32 =");
 #else
-					"SHS-160"
+		    printf("Password SHS-160 =");
 #endif
-						" =");
 		printf(" %08lX", crc);
 	    }
 	}
