--- lilo-22.3.2/mkrescue.orig	Mon Jun 10 04:38:57 2002
+++ lilo-22.3.2/mkrescue	Fri Jul 12 23:13:35 2002
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 #
 # mkrescue
 #
@@ -281,7 +281,7 @@
 
 
 log=$(pwd)/mkrescue.log
-pushd $mount >/dev/null 2>/dev/null
+SAVEPWD="$(pwd)"
 if [ "$fast" = fast ]; then
 	bootcmd="-b /dev/$loopback"
 fi
@@ -293,7 +293,7 @@
 else
 	$lilo -C $mountconfig $compact $bootcmd || fast=error
 fi
-popd >/dev/null 2>/dev/null
+cd "$SAVEPWD"
 if [ "$fast" = error ]; then
 	echo lilo failure.
 else
