--- lilo-22.7.2/diagnose/Makefile.orig	2005-09-23 19:24:33.000000000 +0200
+++ lilo-22.7.2/diagnose/Makefile	2006-07-18 14:07:54.594440000 +0200
@@ -1,11 +1,11 @@
 # Makefile for BCC tests
 #
 MODEL=s
-CC = bcc -M$(MODEL)
+BCC = bcc -M$(MODEL)
 CPP = gcc -E
 CINCL = /usr/lib/bcc/include
 AS = as86
-LINK = $(CC) -v
+LINK = $(BCC) -v
 COPT = -ansi -I$(CINCL) -c -O -DDEBUG=0
 LOPT = -m >$*.map -M -d
 AOPT = -0 -a -w
@@ -17,7 +17,7 @@
 .SUFFIXES:	.com .S
 
 .c.o:
-	$(CC) $(COPT) -o $@ $< -A-l -A$*.lis
+	$(BCC) $(COPT) -o $@ $< -A-l -A$*.lis
 
 .o.com:
 	$(LINK) $(LOPT) -o $@ $^
