diff -urN lilo-22.5.1.orig/Makefile lilo-22.5.1/Makefile
--- lilo-22.5.1.orig/Makefile	Mon Mar 31 14:01:37 2003
+++ lilo-22.5.1/Makefile	Mon Mar 31 14:02:30 2003
@@ -63,8 +63,7 @@
 CFG_DIR=/etc
 BOOT_DIR=/boot
 USRSBIN_DIR=/usr/sbin
-MAN_DIR=`if [ -f /usr/man/manpath ]; then \
-  manpath | sed "s/:.*//"; else echo /usr/man; fi`
+MAN_DIR=/usr/share/man
 #BUILTIN=`./lilo -X | grep CFLAGS | grep LCF_BUILTIN | wc -l | sed "s/ //g"`
 BUILTIN=1
 
